data/README.md

```diff
@@ -0,0 +1,100 @@
+# Data
+
+A guide for managing a series of tubes.
+
+## Best Practices
+
+* Use stream processing for data which is not guaranteed to fit into memory.
+
+### Streaming
+
+* Avoid automatic retries for non-idempotent operations.
+* Avoid storing messages with different schemas in the same topic or message
+  queue.
+* Keep messages as small as possible.
+* Use as few distinct resources (CPU, Postgres, S3, HTTP requests) as possible
+  in each consumer.
+* Use at least once delivery with idempotent operations.
+* Use at most once delivery with non-idempotent operations.
+* Use constant memory to process streams when feasible.
+* When one message would perform more than on operation, instead have that
+  message append a new message for each operation.
+* When reading from the same queue with more than one consumer, ensure that it's
+  acceptable to process messages from that queue out of order.
+* When partitioning messages in a single topic, ensure that it's acceptable for
+  messages from different partitions to be processed out of order.
+
+## Glossary
+
+*Data Engineering*: the task of building infrastructure and process for
+ingesting, processing, and aggregating data so that it can be displayed to users
+or made available to data scientists.
+
+*Data Science*: the practice of using statistics, machine learning, and other
```

Comment:
For those who, like me, didn't understand that stats and ML were different, this one person wrote a (slightly ranty) article explaining the difference: https://towardsdatascience.com/the-actual-difference-between-statistics-and-machine-learning-64b49f07ea3

Reply:
:+1: